src/common/ApprovalsChip.tsx

```diff
@@ -25,11 +42,75 @@ export default function ApprovalsChip({ approvals }: { approvals: any }) {
     backgroundColor = '#4c8a4c'
   }
 
+  const getRequestResponses = (response) => {
+    let Icon
+    let secondaryText
+
+    if (response === 'Accepted') {
+      Icon = Done
+      secondaryText = 'Approved'
+    } else if (response === 'Declined') {
+      Icon = Close
+      secondaryText = 'Declined'
+    } else if (response === 'No Response') {
+      Icon = AccessTime
+      secondaryText = 'Awaiting response'
+    }
+
+    return [Icon, secondaryText]
+  }
+
+  const [ReviewerIcon, reviewerSecondaryText] = getRequestResponses(approvals?.reviewerResponse)
+  const reviewerPrimaryText = `Technical reviewer (${approvals?.reviewer})`
+
+  const [ManagerIcon, managerSecondaryText] = getRequestResponses(approvals?.managerResponse)
+  const managerPrimaryText = `Model manager (${approvals?.manager})`
+
+  const approvalsClicked = (event: MouseEvent) => {
+    setAnchorEl(event.currentTarget as HTMLDivElement)
+  }
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+
   return (
-    <Chip
-      icon={<Icon sx={{ color: 'white !important', pl: 1 }} />}
-      sx={{ borderRadius: 1, color: 'white', height: 'auto', backgroundColor }}
-      label={`Approvals ${numApprovals}/${totalApprovals}`}
-    />
+    <Stack direction='row' spacing={0}>
+      <Chip
+        sx={{ borderRadius: 1, color: 'white', height: 'auto', backgroundColor }}
+        label={`Approvals ${numApprovals}/${totalApprovals}`}
+        onClick={approvalsClicked}
+        icon={
+          open ? (
+            <UpArrow sx={{ color: 'white !important', pl: 1 }} />
+          ) : (
+            <DownArrow sx={{ color: 'white !important', pl: 1 }} />
+          )
+        }
+        aria-controls='model-approvals-menu'
+        aria-haspopup='true'
+        aria-expanded={open ? 'true' : undefined}
+      />
+      <Menu id='model-approvals-menu' anchorEl={anchorEl} open={open} onClose={handleClose}>
+        <List dense={true}>
+          {approvals?.reviewerResponse !== undefined && (
+            <ListItem>
+              <ListItemIcon>
+                <ReviewerIcon aria-hidden={true} />
+              </ListItemIcon>
+              <ListItemText primary={reviewerPrimaryText} secondary={reviewerSecondaryText} />
+            </ListItem>
+          )}
+          {approvals?.managerResponse !== undefined && (
```

Comment:
The `!== undefined` isn't needed as `approvals?.managerResponse` will either be truthy or falsy which is enough in this case. Also `'' !== undefined` returns `true` and so do a lot of other things in JS, so you might get some unintended content being rendered with this check.